x-pack/platform/plugins/shared/spaces/public/space_selector/space_selector.tsx

```diff
@@ -262,6 +272,98 @@ export class SpaceSelector extends Component<Props, State> {
       searchTerm: searchTerm.trim().toLowerCase(),
     });
   };
+
+  public onViewModeChange = (viewMode: 'grid' | 'table') => {
+    this.setState({ viewMode });
+  };
+
+  public getViewToggle = () => {
+    const { spaces, viewMode } = this.state;
+
+    // Only show view toggle if there are spaces to display
+    if (!spaces || spaces.length === 0) {
```

Comment:
nit
```suggestion
    if (!spaces?.length) {
```

though, spaces will always be an array with at least default space, do we even need that check `!spaces`?

Reply:
This code became obsolete and was removed. The toggle now shows within the same logic as the search input (i.e. when there are >8 spaces).